crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -773,7 +773,12 @@ impl<'db> ClassLiteralType<'db> {
                         continue;
                     }
 
-                    if class.is_known(db, KnownClass::Type) && policy.meta_class_no_type_fallback()
+                    // HACK: we should implement some more general logic here that supports arbitrary custom
+                    // metaclasses, not just `type` and `ABCMeta`.
+                    if matches!(
+                        class.known(db),
+                        Some(KnownClass::Type | KnownClass::ABCMeta)
+                    ) && policy.meta_class_no_type_fallback()
```

Comment:
You're robbing @sharkdp of the satisfaction of removing all these ecosystem false positives!

I guess it's also satisfying to remove hacks in the codebase...

Reply:
==> https://github.com/astral-sh/ruff/pull/17733